src/test/java/io/confluent/kafkarest/unit/ConsumerManagerTest.java

```diff
@@ -172,38 +177,52 @@ public void testConsumerNormalOps() throws InterruptedException, ExecutionExcept
     String cid = consumerManager.createConsumer(
         groupName, new ConsumerInstanceConfig(EmbeddedFormat.BINARY));
     sawCallback = false;
+    actualException = null;
+    actualRecords = null;
     consumerManager.readTopic(
         groupName, cid, topicName, BinaryConsumerState.class, Long.MAX_VALUE,
         new ConsumerManager.ReadCallback<byte[], byte[]>() {
           @Override
           public void onCompletion(List<? extends ConsumerRecord<byte[], byte[]>> records,
                                    Exception e) {
+            actualException = e;
+            actualRecords = records;
             sawCallback = true;
-            assertNull(e);
-            assertEquals(referenceRecords, records);
+
+            // Assert that should clearly fail...but doesn't report in test runs
+            assertTrue("This should fail", false);
           }
         }).get();
-    assertTrue(sawCallback);
+    assertTrue("Callback failed to fire", sawCallback);
+    assertNull("No exception in callback", actualException);
+    assertEquals("Records returned not as expected", referenceRecords, actualRecords);
+
     // With # of bytes in messages < max bytes per response, this should finish just after
-    // the per-request timeout (because the timeout perfectly coincides with a scheduled
-    // iteration when using the default settings).
-    assertEquals(config.getInt(KafkaRestConfig.CONSUMER_REQUEST_TIMEOUT_MS_CONFIG)
-                 + config.getInt(KafkaRestConfig.CONSUMER_ITERATOR_TIMEOUT_MS_CONFIG),
-                 config.getTime().milliseconds());
+    // the per-request timeout (attempts to tie this down to the ms not possible so assumes that greater than is "good enough")
+    String msg = "Time taken (" + Long.toString(config.getTime().milliseconds()) + ") to process message should be greater than the timeout " +
+        Integer.toString(config.getInt(KafkaRestConfig.CONSUMER_REQUEST_TIMEOUT_MS_CONFIG)
+                 + config.getInt(KafkaRestConfig.CONSUMER_ITERATOR_TIMEOUT_MS_CONFIG)) ;
+    assertFalse(msg, config.getInt(KafkaRestConfig.CONSUMER_REQUEST_TIMEOUT_MS_CONFIG)
```

Comment:
Because the MockTime now relies on an increment inside the milliseconds method its no longer possible to be exact on time comparison...